test/rekt/resources/broker/broker.go

```diff
@@ -99,6 +103,13 @@ func Install(name string, opts ...CfgFn) feature.StepFn {
 	for _, fn := range opts {
 		fn(cfg)
 	}
+	if dir, ok := cfg["__brokerTemplateDir"]; ok {
+		return func(ctx context.Context, t feature.T) {
+			if _, err := manifest.InstallYaml(ctx, dir.(string), cfg); err != nil {
+				t.Fatal(err)
+			}
+		}
+	}
```

Comment:
I see what you're trying to do here, I had the very same issues when trying to integrate these tests in eventing-kafka-broker.

IMO this is going to be a mess long-term, because it assumes that now resources developers (aka people that develops this install code) has to create smart enough code for having flexible code for the creation. And I'm pretty sure this is going to be a mess, because all the install functions will become inconsistent, undocumented, etc and the final outcome won't be different from the existing testlib. We need to find a solution for this issue at reconciler-test level.

Reply:
Can we have, always through the "install local yaml", swap the template file? Like we have a global map of yamls, and you can for a specific test say "hey i want this template more than the original one".

Or even better, some conf passed through the ctx

Reply:
We are going to start with this, and then develop a way to provide the base version later but. We are still learning and programming for being able to run this in both a vendored and non-vendored way.

Reply:
As @n3wscott said, this is a start, not the final version. For now this IMHO suffices to unblock our ability to test other brokers instead of hardcoding (what's there right now) all over the place. So I'd like to get this merged and we use this as a way to start testing the other brokers and change things as we learn. But, as it stands it's impossible to test other brokers due to hardcoding. @slinkydeveloper fine with this approach?

Reply:
I think we need to stop for a moment and think about how this is going to work, I've already hit this problem several times and IMO we're not really addressing the issue, but just adding hacks around it. We need to reiterate over it and find a proper way to let the _features developer_ care little or nothing about "how this thing is going to be vendored", that's a concern we need to move as much as possible to the framework itself, let's recall that our e2e/conformance tests today are broken and extremely hard to vendor for exactly this reason.
Links to https://github.com/knative-sandbox/reconciler-test/pull/82 which tried to solve a similar/same problem, regarding how one develops a feature and how is then vendored downstream and executed it.
Also this https://github.com/knative/eventing/pull/5072/files#diff-ef47c6d8c23c74e4a54ae1c6cfbea34712ff2a6da2f4bbaf7ec46c97d913900cR46 is another thing that will make these tests hard to be vendorable.

My proposal to move this forward is: can we extract from this PR just the spec mapping features (which is something we need anyway) and remove all the test logic, so we can then figure out both this issue and https://github.com/knative/eventing/pull/5072#discussion_r595079444?

We can start from having `Install` accept some special value in the context to link to a specific yaml, or we can maybe reignite the config pr I did previously? Or maybe should we start thinking to use again in memory data structures for our APIs, aka `Install` accepts eventingv1.Broker data structures directly (and then use the dynamic client to do the actual api invocations)?

Reply:
Let's do it as a followup. We will iterate today, we already have working integrations with RabbitMQBroker, this works and you can vendor it.

Reply:
@vaikas and I have a plan that is being developed and validated here. This needs to land so we can continue to collaborate